src/js/ripple/amount.js

```diff
@@ -574,40 +566,27 @@ Amount.prototype.parse_quality = function(quality, counterCurrency, counterIssue
   this._issuer      = UInt160.from_json(counterIssuer);
   this._is_native   = this._currency.is_native();
 
-  var power = 0;
-  if (this._is_native) {
-    if (opts.inverse) {
-      power += 1;
-    } else {
-      power -= 1;
-    }
+  if (this._is_native && baseCurrency.is_native()) {
+    throw new Error('XRP/XRP quality is not allowed');
   }
 
-  // Correct offset if xrp_as_drops option is not set and base currency is XRP
-  if (!opts.xrp_as_drops &&
-      baseCurrency.is_valid() &&
+  var adjusted;
+  var inversion = opts.inverse ? -1 : 1;
+  // Correct for XRP to drops factors
+  if (this._is_native) {
```

Comment:
Shouldn't xrp_as_drops be honoured regardless of whether it's the base or counter currency ?